relay/blob_manager.go

```diff
@@ -0,0 +1,78 @@
+package relay
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/disperser/common/v2/blobstore"
+	"github.com/Layr-Labs/eigenda/relay/cache"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+// blobManager encapsulates logic for fetching blobs. Utilized by the relay Server.
+// This struct adds caching and threading on top of blobstore.BlobStore.
```

Comment:
It doesn't seem there is any threading below. It's just request limiting.

Reply:
Comment fixed. More accurately, this is not rate limiting, it's instead putting an upper limit on the maximum concurrency for network IO operations. True rate limiting is in a branch that is not yet open for review (will be ready later today, probably).